code/modules/virus2/effect/effect.dm

```diff
@@ -941,6 +941,28 @@ datum/disease2/effect/lubefoot/deactivate(var/mob/living/carbon/mob)
 				mob.apply_damage(10)
 
 
+/datum/disease2/effect/thick_blood
+	name = "Hyper-Fibrinogenesis"
+	stage = 3
+	var/skip = FALSE
+
+/datum/disease2/effect/thick_blood/activate(var/mob/living/carbon/mob)
+	if(skip)
+		return
+	if(ishuman(mob))
+		var/mob/living/carbon/human/H = mob
+		if(H.species && (H.species.anatomy_flags & NO_BLOOD))	//Can't have fragile skin if you don't have skin at all.
+			skip = TRUE
+			return
+	var/mob/living/carbon/human/H = mob
```

Comment:
The rest of this block should be indented under `ishuman()`, otherwise you could end up casting a monkey with the virus as human and attempting to access its limbs, of which it has none.